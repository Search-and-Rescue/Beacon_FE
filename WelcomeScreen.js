import React, { Component } from 'react';
import { View, StyleSheet, Text, Button } from "react-native";

export default class LoginScreen extends Component {
  render() {
    return (
      <View style={{ flex: 1, alignItems: "center", justifyContent: "center" }}>
        <Button
          title="Login"
          onPress={() => this.props.navigation.navigate("Dashboard")}
        />
        <Button title="Sign Up" onPress={() => alert("button pressed")} />
      </View>
    );
  }
}
