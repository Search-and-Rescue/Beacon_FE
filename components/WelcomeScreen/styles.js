import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  loginContainer: {
    alignItems: "center",
    backgroundColor: "#F0F0F0",
    flex: 1,
    justifyContent: "center",
    marginTop: 20
  },
  loginButton: {
    alignItems: "center",
    backgroundColor: "#001028",
    borderColor: "#EFB095",
    borderRadius: 7,
    borderWidth: 4,
    height: 65,
    marginVertical: 10,
    width: 350
  },
  buttonText: {
    color: "#F0F0F0",
    fontFamily: "Futura",
    fontSize: 38,
    paddingVertical: 5
  }
});

export default styles;