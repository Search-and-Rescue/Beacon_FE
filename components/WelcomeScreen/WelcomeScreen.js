import React, { Component } from 'react';
import { View, Text, TouchableOpacity } from "react-native";
import { getEmergencyContacts, getVehicles } from '../../util/apiCalls';
import { setEmergencyContacts, setVehicles } from '../../actions';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import styles from './styles';

export class LoginScreen extends Component {
  componentDidMount = async () => {
    const { setEmergencyContacts, setVehicles } = this.props;
    try {
      const userInfoContacts = await getEmergencyContacts();
      const userInfoVehicles = await getVehicles();
      await setEmergencyContacts(userInfoContacts.user.emergencyContacts);
      await setVehicles(userInfoVehicles.user.vehicles)
    } catch ({ message }) {
      console.log(message)
    }
  }

  render() {
    return (
      <View style={styles.loginContainer}>
        <TouchableOpacity
          style={styles.loginButton}
          title="Login"
          onPress={() => this.props.navigation.navigate("Dashboard")}
        >
          <Text style={styles.buttonText}>Login</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.loginButton}
          title="Login"
          onPress={() => this.props.navigation.navigate("Dashboard")}
        >
          <Text style={styles.buttonText}>Sign Up</Text>
        </TouchableOpacity>
      </View>
    );
  }
}

export const mapDispatchToProps = dispatch => bindActionCreators({ setEmergencyContacts }, dispatch);

export default connect(null, mapDispatchToProps)(LoginScreen);