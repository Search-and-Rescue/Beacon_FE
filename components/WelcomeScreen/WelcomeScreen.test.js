import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import { LoginScreen, mapDispatchToProps } from './WelcomeScreen';
import { configure } from 'enzyme';
import {
  setEmergencyContacts,
  setVehicles,
  setGear,
  setTrips,
  setUser
} from "../../actions";
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('LoginScreen', () => {
  const contactsMock = [{
    name: "Beans",
    phone: "303-1234",
    email: "beans@gmail.com"
  }];
  const vehiclesMock = [{
    id: 1,
    make: 'Ford',
    model: 'F150',
    year: 2010,
    color: 'silver',
    licensePlate: 'CYE 909'
  }];
  const gearMock = [{
    id: 89,
    itemName: 'Rustic Iron Bottle',
    description: 'Chuck Norris can binary search unsorted data.'
  }];
  const userMock = {
    id: 1,
    name: 'Sam Freeman'
  }
  const tripsMock = [{
    name: "Trip name",
    startingPoint: "Starting point",
    endingPoint: "Ending point",
    startDate: "December 1, 2019",
    startTime: "09:00",
    endDate: "December 2, 2019",
    endTime: "09:00",
    notificationDate: "December 3, 2019",
    notificationTime: "09:00",
    travelingCompanions: 0,
    userId: 1
  }]

  it('should match the snapshot with all of the data passed through', () => {
    expect(shallow(<LoginScreen />)).toMatchSnapshot();
  });

  it('it calls dispatch with the setEmergencyContacts action', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = setEmergencyContacts(contactsMock);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.setEmergencyContacts(contactsMock);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it('it calls dispatch with the setVehicles action', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = setVehicles(vehiclesMock);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.setVehicles(vehiclesMock);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it('it calls dispatch with the setGear action', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = setGear(gearMock);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.setGear(gearMock);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it('it calls dispatch with the setUser action', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = setUser(userMock);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.setUser(userMock);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it('it calls dispatch with the setTrips action', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = setTrips(tripsMock);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.setTrips(tripsMock);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});