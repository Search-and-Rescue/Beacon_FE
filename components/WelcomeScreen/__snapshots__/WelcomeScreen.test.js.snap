// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen should match the snapshot with all of the data passed through 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F0F0F0",
      "flex": 1,
      "justifyContent": "center",
      "marginTop": 20,
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#001028",
        "borderColor": "#EFB095",
        "borderRadius": 7,
        "borderWidth": 4,
        "height": 65,
        "marginVertical": 10,
        "width": 350,
      }
    }
    title="Login"
  >
    <Text
      style={
        Object {
          "color": "#F0F0F0",
          "fontFamily": "Futura",
          "fontSize": 38,
          "paddingVertical": 5,
        }
      }
    >
      LOGIN
    </Text>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#001028",
        "borderColor": "#EFB095",
        "borderRadius": 7,
        "borderWidth": 4,
        "height": 65,
        "marginVertical": 10,
        "width": 350,
      }
    }
    title="Login"
  >
    <Text
      style={
        Object {
          "color": "#F0F0F0",
          "fontFamily": "Futura",
          "fontSize": 38,
          "paddingVertical": 5,
        }
      }
    >
      SIGN UP
    </Text>
  </TouchableOpacity>
</View>
`;
